pkg/controller/stacks/hosted/config.go

```diff
@@ -59,14 +70,30 @@ func NewConfig(hostControllerNamespace, tenantAPIServiceHost, tenantAPIServicePo
 	}, nil
 }
 
-// ObjectReferenceOnHost maps object with given name and namespace into single controller namespace on Host Cluster.
+// ObjectReferenceOnHost maps object with given name and namespace into single
+// controller namespace on Host Cluster.
+// The resource name on the host cluster is truncated to label value length
```

Comment:
Yes. I'll defer a deeper explanation to the other review comments. In short, a 253 character tenant resource name (505 with namespace) needs to be representable in a 63 character host label value.

Reply:
And that 63 character name should be predictable based on the tenant (source) resource name.

Reply:
While adding some debug examples, I learned namespaces can only be 63 characters. ~505~ 315.

Reply:
I discovered some caveats to the name length restrictions while doing some testing in response to @jbw976's request for debugging examples.

Namespaces can not exceed 63 characters. 
Deployments and Jobs, the resources that we intend to create on the host, are limited to 63 characters because their names are used in label values.

Of note for #1354, Secrets (as well as ServiceAccounts) abide by the 253 character limit (with 63 character labels) described in https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

Stack and StackInstall don't seem to abide by this naming policy as I was able to create one with 70+ characters in the name, without dots.  Since Stack and StackInstall resources generate deployments and jobs with matching names, they are limited, in practice to 63 characters, which this PR addresses for both hosted and local stack deployments and jobs.